container-insecure-registry/insecure-registry-config.yaml

```diff
@@ -0,0 +1,43 @@
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: insecure-registries
+  namespace: default
+  labels:
+    k8s-app: insecure-registries
+spec:
+  selector:
+    matchLabels:
+      name: insecure-registries
+  updateStrategy:
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        name: insecure-registries
+    spec:
+      nodeSelector:
+        cloud.google.com/gke-container-runtime: "containerd"
+      hostPID: true
+      containers:
+        - name: startup-script
+          image: gcr.io/google-containers/startup-script:v1
+          imagePullPolicy: Always
+          securityContext:
+            privileged: true
+          env:
+          - name: STARTUP_SCRIPT
+            value: |
+              set -o errexit
+              set -o pipefail
+              set -o nounset
+
+              export REGISTRY_ADDRESS=ADDRESS
```

Comment:
where the `ADDRESS` is coming from?

Reply:
two things to consider:

Add an explicit 

```
env:
    - name: ADDRESS
      value: "Hello from the environment"
```

section to the pod spec

Or make the bash script fail if the env variable is not set.

Reply:
I guess it will already fail complaining that `ADDRESS` is not a function. But perhaps it can be more elabiorate check.

But something like commented out export and this after the comment would work the best I think:

```
if [ -z "$REGISTRY_ADDRESS" ]; then
  echo "Error: Environment variable REGISTRY_ADDRESS is not set."
  exit 1
fi
```

Reply:
I changed it to look for either empty or the placeholder value and throw an error, but I'm not sure how to test the daemonset for success - running it in a GKE cluster showed no logs in any of the Pods